src/app/auth/auth-direct-login/auth-direct-login.component.ts

```diff
@@ -85,6 +85,13 @@ export class AuthDirectLoginComponent implements OnInit {
       // switch to the program
       await this.switcherService.switchProgram(program).toPromise();
     }
+    const referrerUrl = this.route.snapshot.paramMap.get('activity_task_referrer_url');
```

Comment:
@jazzmind `activity_task_referrer_url` is meant to be the referrer URL for direct link to the activity_task page.
If we need the similar logic for assessment page, shall we name it `assessment_referrer_url` and put it under the assessment `case` accordingly?

Reply:
I think we should have a generic referrer URL for any direct link which is where the back arrow and similar behavior goes. It is not specific to the activity_task so it should just be referrer_url... however we should leave activity_task_referrer_url in there as an alternative for backward compatiblity.

Reply:
The problem is how do we know which back button should be redirected to the referrer URL
e.g. we may have a use case that we want a user to navigate the activity page and assessment/topic page in app, only redirect user back to referrer when user click back button from activity page.
we may have another use case that we want a user to only see the assessment page in app, redirect user back to referrer when user click back button from assessment page.
We will need a way to handle this different scenario. That's why I put this activity_task referrer URL to the "activity task" specific case

Reply:
Sounds like we need some design work here to list out use cases we want to support for embedding the app. I agree in the interim to call it assessment_referrer_url so that we preserve optionality.

Reply:
I have made a change which i think gives us some flexibility and a level of abstraction... let me know what you think @shawnm0705